### PR TITLE
KFSPTS-24454: Backport FINP-8322 improve performance of action request update for responsibility change

### PR DESCRIPTION
**Pull request by nkimble**

### CODE REVIEW

File: src/main/java/org/kuali/kfs/kew/actionrequest/service/ActionRequestService.java [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+
+    List<ActionRequest> findByStatusAndDocId(String statusCd, String documentId);
+
+    void alterActionRequested(List actionRequests, String actionRequestCd);
+
+    List<ActionRequest> findByDocumentIdIgnoreCurrentInd(String documentId);
+
+    List findActivatedByGroup(String groupId);
+
+    void updateActionRequestsForResponsibilityChange(Set<String> responsibilityIds);
```
Comment (jhulslander) [inline src/main/java/org/kuali/kfs/kew/actionrequest/service/ActionRequestService.java:115]: The KualiCo commit adds comments to this line.  Should they also be added?

/**
     * Schedule requeue of documents, affected by responsibility changes.
     *
     * @param responsibilityIds used to find documents to requeue.
     */

**Review by jhulslander — COMMENTED** [review]
These changes look good.  I did notice one javadoc item added in the kualico commit that wasn't in this PR.  Should it be added here, or do we not need it.

File: src/main/java/org/kuali/kfs/kew/actionrequest/service/ActionRequestService.java (hunk 1 above)
Comment (nkimble) [reply src/main/java/org/kuali/kfs/kew/actionrequest/service/ActionRequestService.java:120]: comment added.

**Review by jhulslander — APPROVED** [review]
my concerns have been addressed, I think can be merged

**Pull request merged**